File: replace_logo_script.py
import os
import re

directory = r"c:\Users\sabarinath\Desktop\developer\therapy"

# HTML logo pattern: `<i class="bi bi-chat-heart-fill"></i>\s*<span>SpeechCare</span>`
pattern = re.compile(r'<i class="bi bi-chat-heart-fill"></i>\s*<span>SpeechCare</span>', re.IGNORECASE)
replacement = '<img src="assets/images/logo.svg" alt="Kinderly Logo" height="40">'

count = 0
for filename in os.listdir(directory):
    if filename.endswith(".html"):
        filepath = os.path.join(directory, filename)
        with open(filepath, "r", encoding="utf-8") as f:
            content = f.read()
        
        new_content, num_subs = pattern.subn(replacement, content)
        
        if num_subs > 0:
            with open(filepath, "w", encoding="utf-8") as f:
                f.write(new_content)
            print(f"Updated {filename} ({num_subs} replacements)")
            count += 1

print(f"Total files updated: {count}")

File: replace_logo_script.js
const fs = require('fs');
const path = require('path');

const dir = 'c:\\Users\\sabarinath\\Desktop\\developer\\therapy';
const files = fs.readdirSync(dir).filter(f => f.endsWith('.html'));

const regex = /<i class="bi bi-chat-heart-fill"><\/i>\s*<span>SpeechCare<\/span>/gi;
const replacement = '<img src="assets/images/logo.svg" alt="Kinderly Logo" height="40">';

let count = 0;
files.forEach(file => {
    const filePath = path.join(dir, file);
    const content = fs.readFileSync(filePath, 'utf8');
    
    if (regex.test(content)) {
        const newContent = content.replace(regex, replacement);
        fs.writeFileSync(filePath, newContent, 'utf8');
        console.log(`Updated ${file}`);
        count++;
    }
});

console.log(`Total files updated: ${count}`);
